src/main/webapp/app/exercises/quiz/manage/drag-and-drop-question/drag-and-drop-question-edit.component.ts

```diff
@@ -875,4 +887,112 @@ export class DragAndDropQuestionEditComponent implements OnInit, OnChanges, Afte
         this.cleanupQuestion();
         this.markdownEditor.parse();
     }
+
+    /**
+     * Create new drag items for each drop location in the background image
+     * @returns
+     */
+    getImagesFromDropLocations() {
+        for (const someLocation of this.question.dropLocations!) {
+            const image = new Image();
+            let dataUrl: string = '';
+            let bgWidth;
+            let bgHeight;
+            image.onload = () => {
+                bgHeight = image.height;
+                bgWidth = image.width;
+
+                const canvas = document.createElement('canvas');
+                const context = canvas.getContext('2d');
+
+                if (context) {
+                    const scalarHeight = bgHeight / 200;
+                    const scalarWidth = bgWidth / 200;
+                    canvas.width = someLocation.width! * scalarWidth;
+                    canvas.height = someLocation.height! * scalarHeight;
+                    context.drawImage(
+                        image,
+                        someLocation.posX! * scalarWidth,
+                        someLocation.posY! * scalarHeight,
+                        someLocation.width! * scalarWidth,
+                        someLocation.height! * scalarHeight,
+                        0,
+                        0,
+                        someLocation.width! * scalarWidth,
+                        someLocation.height! * scalarHeight,
+                    );
+
+                    dataUrl = canvas.toDataURL('image/png');
+                    const dragItemCreated = this.createImageDragItemFromFile(this.dataUrlToFile(dataUrl, 'placeholder' + someLocation.posX!))!;
+                    const dndMapping = new DragAndDropMapping(dragItemCreated, someLocation);
+                    this.question.correctMappings!.push(dndMapping);
+                }
+            };
+            image.src = this.backgroundImage.src;
+        }
+        this.blankOutBackgroundImage();
+    }
+
+    /**
+     * Takes all drop locations and replaces their location with a white rectangle on the background image
+     */
+    blankOutBackgroundImage() {
+        const backGroundBlankingCanvas = document.createElement('canvas');
+        const backGroundBlankingContext = backGroundBlankingCanvas.getContext('2d');
+        const image = new Image();
+        let bgWidth;
+        let bgHeight;
+        image.onload = () => {
+            bgHeight = image.height;
+            bgWidth = image.width;
+
+            backGroundBlankingCanvas.width = bgWidth;
+            backGroundBlankingCanvas.height = bgHeight;
+            if (backGroundBlankingContext) {
+                const scalarHeight = bgHeight / 200;
+                const scalarWidth = bgWidth / 200;
```

Comment:
What's the meaning of 200 here?

Reply:
The brackground is scaled to a 200x200 click layer for the drop locations .

Reply:
Makes sense - perhaps it would make sense to define a constant for this since you use it in two locations `blankOutBackgroundImage` and `getImagesFromDropLocations`. 

(Optional: Those two methods look very similar, which is why I assume you've placed the comment in the other method rather than in the spot I marked. Is it possible to refactor parts of these similar-looking methods?)